File: 2026POTD/FEBRUARY/07-02-2026/1653.java
class Solution {
    public int minimumDeletions(String s) {

        int ans=0;
           int count=0;
        for(int i=0;i<s.length();i++){
            if(s.charAt(i)=='b'){
                count++;
            }
            if(s.charAt(i)=='a' && count>0){
                count--;
                ans++;
            }
            
             
        }
        return ans;
    }
}

File: 2026POTD/FEBRUARY/07-02-2026/1653.cpp
class Solution {
public:
    int minimumDeletions(string s) {
        stack<char> stk;int count=0;
        for(char ch:s){
            if(!stk.empty() && stk.top()=='b' && ch=='a'){
                stk.pop();
                count++;
            }
            else{
                stk.push(ch);
            }
        } 
        return count;
    }
};